playbooks/roles/deploy-osh/tasks/helm-run.yml

```diff
@@ -1,63 +1,51 @@
 ---
 # Inspired by https://github.com/openstack/openstack-helm-infra/blob/master/roles/build-helm-packages/tasks/setup-helm-serve.yaml
-- block:
-    - name: checking if local helm server is running
-      wait_for:
-        delay: 5
-        timeout: 10
-        host: 127.0.0.1
-        port: 8879
-        state: started
-        msg: "Failure to connect on helm server -- Attempting automatic recovery"
-      register: _server_running
+- name: Check if systemd unit file is present
+  stat:
+    path: /etc/systemd/system/helm-serve.service
+  register: helm_serve_service
 
-    - name: Check port 8879 is running a helm repository
-      uri:
-        url: http://127.0.0.1:8879
-        return_content: yes
-      register: this
-      failed_when: "'Helm Repository' not in this.content"
+- name: Copy systemd unit into place for helm server if it is not present
+  when: not helm_serve_service.stat.exists
+  template:
+    src: helm-serve.service.j2
+    dest: /etc/systemd/system/helm-serve.service
+    mode: 0640
 
-  rescue:
-    - name: Fixing previous failure - moving systemd unit into place for helm server
-      template:
-        src: helm-serve.service.j2
-        dest: /etc/systemd/system/helm-serve.service
-        mode: 0640
+- name: Ensure helm serve service is running
+  systemd:
+    state: started
+    daemon_reload: yes
+    name: helm-serve
+    enabled: yes
 
-    - name: Fixing previous failure - starting helm serve service
-      systemd:
-        state: restarted
-        daemon_reload: yes
-        name: helm-serve
-        enabled: yes
+- name: Check if helm server is up
+  wait_for:
+    delay: 5
+    host: 127.0.0.1
+    port: 8879
+    state: started
 
-    - name: Check if helm server is up now
-      wait_for:
-        delay: 5
-        host: 127.0.0.1
-        port: 8879
-        state: started
+- name: Check port 8879 is running a helm repository
+  uri:
+    url: http://127.0.0.1:8879
+    return_content: yes
+  register: this
+  failed_when: "'Helm Repository' not in this.content"
```

Comment:
Is there a race between the helm server listening on 8879 and serving the correct content? Should we have a retry/timeout here?

Also, previously when this check failed the service was restarted and appeared to work. This might have been in place as a workaround for something we're unaware of. We should perhaps leave a comment here or look at a similar "restart on failure".

Reply:
But the only way the original rescue worked was to provide the unit file and start the service again. Now that I make sure the service file is present, I do not see another reason for repair - if it still does not work, user should know